app/src/test/java/com/mho/training/features/main/MainViewModelTest.kt

```diff
@@ -0,0 +1,339 @@
+package com.mho.training.features.main
+
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
+import androidx.lifecycle.Observer
+import com.example.android.domain.Movie
+import com.example.android.domain.result.DataResult
+import com.example.android.testshared.mockedMovie
+import com.example.android.usecases.*
+import com.mho.training.enums.MovieCategoryEnum
+import com.mho.training.utils.Event
+import com.nhaarman.mockitokotlin2.atLeast
+import com.nhaarman.mockitokotlin2.given
+import com.nhaarman.mockitokotlin2.verify
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.runBlocking
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.TestRule
+import org.junit.runner.RunWith
+import org.mockito.Mock
+import org.mockito.junit.MockitoJUnitRunner
+import java.io.IOException
+
+/*
+[functionName]Should[Call|Return|Throw|Insert|etc][Function|Object|Exception][WithGiven][When]
+ */
+
+@RunWith(MockitoJUnitRunner::class)
+class MainViewModelTest {
+
+    @get:Rule
+    val rule: TestRule = InstantTaskExecutorRule()
+
+    @Mock lateinit var getFavoriteMovieListUseCase: GetFavoriteMovieListUseCase
+    @Mock lateinit var getFavoriteMovieListWithChangesUseCase: GetFavoriteMovieListWithChangesUseCase
+    @Mock lateinit var getPopularMovieListUseCase: GetPopularMovieListUseCase
+    @Mock lateinit var getTopRatedMovieListUseCase: GetTopRatedMovieListUseCase
+    @Mock lateinit var getInTheatersMovieListUseCase: GetInTheatersMovieListUseCase
+
+    @Mock lateinit var observerLoading: Observer<Boolean>
```

Comment:
Grammatically it could be better:
```suggestion
    @Mock lateinit var loadingObserver: Observer<Boolean>
```